drivers/serial/uart_ns16550.c

```diff
@@ -358,7 +427,8 @@ static uint8_t ns16550_inbyte(const struct uart_ns16550_device_config *cfg,
 	return 0;
 }
 
-#if UART_NS16550_PCP_ENABLED
+#if (defined(CONFIG_UART_NS16550_INTEL_LPSS_DMA) & (defined(CONFIG_UART_ASYNC_API)))\
```

Comment:
IIRC, LPSS supports PCP, so it is better to flow this into `UART_NS16550_PCP_ENABLED`.

Reply:
UART_NS16550_PCP_ENABLED is enabled only if any of the UART instances has pcp property in dts, on RPL it is not enabled. This function will through warning as unused if the dependency on CONFIG_UART_NS16550_INTEL_LPSS_DMA and CONFIG_UART_ASYNC_API is not added.